documentation/optee_design.md

```diff
@@ -404,180 +239,165 @@ thread_svc_regs` in case the TA is executed in AArch64.
 
-When returning `SP_EL0` is assigned what's in `struct thread_svc_regs`.
-This allows `tee_svc_sys_return_helper()` having the syscall exception
-handler return directly to `thread_unwind_user_mode()`.
+When returning `SP_EL0` is assigned what's in `struct thread_svc_regs`. This
+allows `tee_svc_sys_return_helper()` having the syscall exception handler return
+directly to `thread_unwind_user_mode()`.
 
 # 7. Shared Memory
-
-Shared Memory is a block of memory that is shared between the non-secure
-and the secure world. It is used to transfer data between both worlds.
+Shared Memory is a block of memory that is shared between the non-secure and the
+secure world. It is used to transfer data between both worlds.
 
 ## Shared Memory Allocation
-The shared memory is allocated by the Linux driver from a pool
-`struct shm_pool`. The pool contains:
-* the physical address of the start of the pool
-* the size of the pool
-* whether or not the memory is cached
-* list of chunk of memory allocated.
-
-Note that
-- the shared memory pool is physically contiguous.
-- the shared memory area is not protected as it is used by both
-  non-secure and secure world.
+The shared memory is allocated by the Linux driver from a pool `struct
+shm_pool`, the pool contains:
+* The physical address of the start of the pool
+* The size of the pool
+* Whether or not the memory is cached
+* List of chunk of memory allocated.
+
+Note that:
+- The shared memory pool is physically contiguous.
+- The shared memory area is not protected as it is used by both non-secure and
+  secure world.
 
 ### Shared Memory Configuration
 It is the Linux kernel driver for OP-TEE that is responsible for initializing
-the shared memory pool,
-given information provided by the Trusted OS. The Linux driver issues a SMC call
-OPTEE_SMC_GET_SHM_CONFIG to retrieve the information
-* physical address of the start of the pool
-* size of the pool
-* whether or not the memory is cached
+the shared memory pool, given information provided by the Trusted OS. The Linux
+driver issues a SMC call `OPTEE_SMC_GET_SHM_CONFIG` to retrieve the information
+* Physical address of the start of the pool
+* Size of the pool
+* Whether or not the memory is cached
 
 The shared memory pool configuration is platform specific. The memory mapping,
-including the area MEM_AREA_NSEC_SHM (shared memory with non-secure world), is
+including the area `MEM_AREA_NSEC_SHM` (shared memory with non-secure world), is
 retrieved by calling the platform-specific function `bootcfg_get_memory()`.
-Please refer to this function and the area type MEM_AREA_NSEC_SHM to see
-the configuration for the platform of interest.
-
-The Linux driver will then initialize the shared memory pool accordingly.
+Please refer to this function and the area type `MEM_AREA_NSEC_SHM` to see the
+configuration for the platform of interest. The Linux driver will then
+initialize the shared memory pool accordingly.
 
 ### Shared Memory Chunk Allocation
```

Comment:
I agree, that was a part of what I had in mind with updates in general regarding changes to shared memory lately. There is both the work you've done (the extensions) and then @lorc did some work just recently.